test/client.test.js

```diff
@@ -143,6 +147,97 @@ describe('Client', () => {
     return expect(resultWithoutOptions).toEqual(resultWithOptions)
   })
 
+  test('Client#close call on Http2Adapter-based Client', async () => {
+    const client = util.getClient({
+      http2SessionIdleTime: Infinity,
+    })
+
+    await client.ping()
+
+    expect(client._http._adapter._closed).toBe(false)
+
+    await client.close()
+
+    expect(client._http._adapter._closed).toBe(true)
+    expect(
+      Object.values(client._http._adapter._sessionMap).every(
+        ({ session }) => session.closed
+      )
+    ).toBe(true)
+
+    await expect(client.ping()).rejects.toThrow(
+      'The Client has already been closed'
+    )
+  })
+
+  test('Client#close call on Http2Adapter-based Client with force=true', async () => {
+    const client = util.getClient()
+
+    await expect(
+      Promise.all([
+        client.ping(),
+        client.close({
+          force: true,
+        }),
+      ])
+    ).rejects.toThrow('The request is aborted due to the Client#close call')
```

Comment:
Is there any way to check the class (`ClientClosed`) rather than the error message? It would be nice if the test did not depend on the particular phrasing of the message.